### ISSUE TITLE
Toward scikit-activeml for openml

### ISSUE THREAD

**Issue by mherde** [scikit-activeml/scikit-activeml#272]
This issue serves as a starting point for a discussion on whether and how [scikit-activeml](https://scikit-activeml.github.io/scikit-activeml-docs/) can be used to perform active learning evaluation via the [openml](https://www.openml.org/) platform. Under the assumption scikit-activeml can be combined with openml, the final **goal** is to create a feature request according to the [openml template](https://github.com/orgs/openml/discussions/categories/openml-design-and-feature-requests) for task types regarding active learning. 

For this purpose, we invite interested people to comment on the topic of evaluating active learning strategies in general and, in particular, in the context of scikit-activeml and openml.  In favor of a more structured and dedicated discussion, our current focus lies on the following aspects:
- pool-based active learning,
- classification,
- omniscient annotators.

We will update this initial comment from time to time to summarize the key insights throughout the discussion.

**Comment by mherde** [comment]
In the following, a list of resources for understanding the workflow of openml is given.
- openml relies on a number of [core concepts](https://docs.openml.org/), i.e., datasets, tasks, flows, and runs.
- [Tasks](https://docs.openml.org/#tasks), as a central tool for integrating active learning into openml, describe how to deal with data. Each task is designed to a specific task type. Currently, openml supports [eight task types](https://www.openml.org/search?type=task_type&sort=date), e.g., supervised classification and regression.

**Comment by mherde** [comment]
Evaluating **active learning** substantially differs from evaluating "traditional" supervised learning. In the following, we try to provide a list of aspects to be taken into account when assessing performances in active learning and indicate which ones are novel [N].
- We evaluate a _query strategy_  [N] in combination with a _target model_ (e.g., classifier, regressor) and optionally a _data model_  [N] (e.g., epistemic uncertainty estimator).
- In each active learning cycle, we quantify the performance of the target model on a test set using a corresponding _evaluation measure_ (e.g., accuracy, mean squared error).
- Given the test performance of each active learning cycle, there are varying _aggregation techniques_  [N] for comparing active learning methods. Examples are:
   - Comparing the performances after a predefined amount of annotation cost. 
   - Comparing the areas under learning curves by meaning the performances across the active learning cycles.
 - The _train-test split method_ serves to separate the train from test data, where the train data serves as a pool of samples for which labels can be acquired.
 -

**Comment by mherde** [comment]
Technical aspects to consider:
- `extensions/functions.py: get_extension_by_model` calls `can_handle_model`, which is fulfilled the scikit-learn extension in combination with all scikit-activeml. For active learning experiments, our model will be tuple `(query_strategy, prediction_model)` or triple `(query_strategy, prediction_model, selector_model)` such that only our own extension for openml can handle it.

**Comment by tpham93** [comment]
A concrete development of a prototypical scikit-activeml extension toward openml can be found here:  https://github.com/mherde/openml-python
